text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
+
+**Environment variables**:
+
+* `vars` (renamed from `env`): yields a vector of `(OsStrBuf, OsStrBuf)` pairs.
```

Comment:
It seems a little unfortunate to convert from a `Vec` to an `Iterator` when we started with a `Vec`: if someone actually needs a `Vec`, there'll be two allocations, the original one and the `collect`d one. (This applies to the `args` case somewhat too, since many platforms start with a `Vec` forcing two allocations there.)